src/HttpLib/HttpRequestHandler.java

```diff
@@ -43,7 +44,12 @@ public HttpResponse send(HttpRequest request) throws InvalidRequestException, IO
         in.close();
         socket.close();
 
-        return new HttpResponse(res.toString());
+        try {
+            return new HttpResponse(res.toString());
+        } catch (HttpFormatException e) {
+            // Invalid response format received, sending invalid empty reponse
+            return new HttpResponse();
```

Comment:
Some sort of error should be thrown here, catching exceptions and doing nothing with them will lead to harder debugging.

Reply:
It depends on how we want the handler to act/react to it. Should it handle everything (like right now) and return an empty response, or should it throw directly and let the application deal with a wrongly formatted response ?